File: pants-mall-backend/mall/src/main/java/com/pants/mall/entity/Review.java
package com.pants.mall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@TableName("review")
@EqualsAndHashCode(callSuper = true)
public class Review extends BaseEntity {
    private Long id;
    private Long userId;
    private Long orderId;
    private Long spuId;
    private Long skuId;
    private Integer rating;
    private String content;

    /**
     * 尺码感受：偏小 / 合适 / 偏大
     */
    private String sizeFeel;

    /**
     * 裤长感受：偏短 / 合适 / 偏长
     */
    private String lengthFeel;

    /**
     * 版型感受：修身 / 合适 / 宽松
     */
    private String fitFeel;

    /**
     * 面料感受：偏硬 / 适中 / 偏软
     */
    private String fabricFeel;

    /**
     * 购买尺码
     */
    private String purchaseSize;

    /**
     * 是否匿名：0否 1是
     */
    private Integer anonymous;

    /**
     * 评价图片，多个图片 URL 使用英文逗号分隔
     */
    private String images;
}

File: pants-mall-frontend/src/api/review.ts
import client from './client'

type Result<T> = {
  code: number
  msg: string
  data: T
}

export type ReviewCreateReq = {
  orderId: string | number
  spuId: string | number
  skuId: string | number
  rating: number
  content: string
  sizeFeel?: string
  lengthFeel?: string
  fitFeel?: string
  fabricFeel?: string
  purchaseSize?: string
  anonymous?: number
  images?: string[]
}

export type ReviewItemVO = {
  id?: string | number
  userId?: string | number
  orderId?: string | number
  spuId?: string | number
  skuId?: string | number
  rating?: number
  content?: string
  sizeFeel?: string
  lengthFeel?: string
  fitFeel?: string
  fabricFeel?: string
  purchaseSize?: string
  anonymous?: number
  username?: string
  images?: string[]
  createdAt?: string
}

export async function createReview(data: ReviewCreateReq) {
  const resp = await client.post<Result<string>>('/review/create', data)
  return resp?.data
}

export async function listReviewsBySpuId(spuId: string | number) {
  const resp = await client.get<Result<ReviewItemVO[]>>('/review/list', {
    params: { spuId },
  })
  return resp?.data?.data || []
}